docs/DeveloperGuide.adoc

```diff
@@ -96,171 +109,383 @@ The `UI` component,
-* **Alternative 2:** Individual command knows how to undo/redo by itself.
-** Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
-** Cons: We must ensure that the implementation of each individual command are correct.
+* **Alternative 1 (current choice):** Create a new `DailyFoodLog` to pass into `ModelManager` and `FoodRecord`.
+** Pros: Maintain comprehensive layers of abstraction and allows code to be easily testable.
+** Cons: Difficult for newcomers or even existing users to trace because of long execution path.
 
-===== Aspect: Data structure to support the undo/redo commands
+* **Alternative 2:** Bypass `ModelManager` or even not use `FoodRecord` for storage of data during runtime by allowing everything to be done
+from parser.
+** Pros: Reduce dependencies on `ModelManager` and `FoodRecord`, and make code contained in a single class file easier to navigate.
+** Cons: Violates layers of abstraction set in place by previous structure of AddressBook3. Violates Single Responsibility Principle and reduce
+cohesiveness of code.
 
-* **Alternative 1 (current choice):** Use a list to store the history of address book states.
-** Pros: Easy for new Computer Science student undergraduates to understand, who are likely to be the new incoming developers of our project.
-** Cons: Logic is duplicated twice. For example, when a new command is executed, we must remember to update both `HistoryManager` and `VersionedAddressBook`.
-* **Alternative 2:** Use `HistoryManager` for undo/redo
-** Pros: We do not need to maintain a separate list, and just reuse what is already in the codebase.
-** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
-// end::undoredo[]
+===== Aspect: Data structure to support the consumption commands
+
+* **Alternative 1 (current choice):** Use a single `FilteredList` to store food for any day by repopulating it each time
+a consumption related command is used.
+** Pros: Only uses a single `FilteredList`, so it is clear which list you are using for display.
+** Cons: May have performance issue in terms of speed when there are too many entries.
+
+* **Alternative 2:** Use a `FilteredList` for each date, to store food consumed on that date.
+** Pros: Faster retrieval for display of `ConsumedFood` items. However, under practical circumstances, the difference is negligible.
+** Cons: May have performance issue in terms of storage because it requires many lists to be stored in addition to `LinkedHashMap`
+in `DailyFoodLog` for each `LocalDate`.
+
+==== Summary
+
+The `nom` command adds a `Food` item consumed by the user into the `stomach`. The following activity diagram summarizes what happens when the user executes a `nom` command.
+
+.Activity Diagram for Nom
+image::NomActivityDiagram.png[]
```

Comment:
The "User executes Nom command" box has already stated that it's a Nom command before the branch conditional "[command is Nom]".